packages/engine/src/proto.rs

```diff
@@ -226,9 +227,27 @@ pub struct ExperimentRunBase {
 }
 
 impl ExperimentRunBase {
-    /// Returns `true` if the experiment uses Python init or has any Python behavior.
-    pub fn requires_python_runner(&self) -> bool {
-        self.project_base.package_init.initial_state.name == InitialStateName::InitPy
+    /// Returns a `RunnerSpawnConfig` matching the config required by the files present in the
+    /// experiment.
+    pub fn runner_spawn_config(&self) -> RunnerSpawnConfig {
+        RunnerSpawnConfig {
+            python: self.requires_runner(Language::Python),
+            rust: self.requires_runner(Language::Rust),
+            javascript: self.requires_runner(Language::JavaScript),
+        }
+    }
+
+    /// Returns `true` if the experiment uses the language's init or has any behavior of the
+    /// language.
+    fn requires_runner(&self, language: Language) -> bool {
+        let requires_init = match (language, &self.project_base.package_init.initial_state.name) {
+            (Language::JavaScript, InitialStateName::InitJson) => true,
```

Comment:
I'm not sure if `InitJson` makes the JS runner required, I guessed it was.

Reply:
My guess is that's not required. Could you make a commit removing this line and check if CI fails?

Reply:
CI is ok with it https://github.com/hashintel/hash/pull/552/checks?sha=b738148d03fe60e44e72bf84b1df9de264e54ace